src/sortStudents.ts

```diff
@@ -1,16 +1,68 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'Name',
+  Surname = 'Surname',
+  Age = 'Age',
+  Married = 'Married',
+  AverageGrade = 'Grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy = [...students];
+  const calculateAverageGrade = (grades: number[]): number => {
+    return grades.reduce((sum, grade) => sum + grade, 0) / grades.length;
+  };
+
+  studentsCopy.sort((a, b) => {
+    let comparison = 0;
+
+    switch (sortBy) {
+      case SortType.Name:
+        comparison = a.name.localeCompare(b.name);
+        break;
+      case SortType.Surname:
+        comparison = a.surname.localeCompare(b.surname);
+        break;
+      case SortType.Age:
+        comparison = a.age - b.age;
+        break;
+      case SortType.Married:
+        if (a.married === b.married) {
+          comparison = 0;
+          break;
+        }
+
+        if (a.married) {
+          comparison = 1;
+          break;
+        }
+
+        comparison = -1;
```

Comment:
this can be simply put like that also because `Number(true) === 1` and `Number(false) === 0`
```suggestion
       comparison = Number(a.married) - Number(b.married);
```